File: sdks/csharp/src/NovaBilling/Charges/Requests/ListChargesRequest.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record ListChargesRequest
{
    /// <summary>
    /// Filter by plan ID
    /// </summary>
    [JsonIgnore]
    public string? PlanId { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/php/src/Charges/Requests/ListChargesRequest.php
<?php

namespace NovaBilling\Charges\Requests;

use NovaBilling\Core\Json\JsonSerializableType;

class ListChargesRequest extends JsonSerializableType
{
    /**
     * @var ?string $planId Filter by plan ID
     */
    public ?string $planId;

    /**
     * @param array{
     *   planId?: ?string,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->planId = $values['planId'] ?? null;
    }
}
